app/src/androidTest/java/org/simple/clinic/storage/PurgeDatabaseAndroidTest.kt

```diff
@@ -458,4 +458,52 @@ class PurgeDatabaseAndroidTest {
     assertThat(patientDao.patientProfileImmediate(patientWithNotPassedRetentionTime.patientUuid)).isEqualTo(patientWithNotPassedRetentionTime)
     assertThat(patientDao.patientProfileImmediate(patientWithNotPassedRetentionTimeAndDeleteAtNull.patientUuid)).isEqualTo(patientWithNotPassedRetentionTimeAndDeleteAtNull)
   }
+
+  @Test
+  fun purging_the_database_should_delete_soft_deleted_blood_pressure_meansurements_when_patient_is_null() {
```

Comment:
Same comment as before, are we going to only remove data that is synced and soft deleted? You might have carried over conditions from other purge queries, please confirm that once.